core/src/main/java/hudson/Proc.java

```diff
@@ -64,6 +64,11 @@
 public abstract class Proc {
     protected Proc() {}
 
+    /**
+     * Indicates whether the process should be killed on interruption.
+     */
+    protected boolean killWhenInterrupted = true;
```

Comment:
I would suggest using `final private` fields and getters

Reply:
It can't be `final` since the value is set in the implementations.
Also it can't be `private` in order to be accessible from implementations. Maybe I should add a setter and make it private?